test/lib/mayaUsd/utils/test_ShaderGenUtils.cpp

```diff
@@ -151,3 +153,54 @@ TEST(ShaderGenUtils, topoGraphAPI_defaultgeomprop)
     }
 }
 #endif
+
+TEST(ShaderGenUtils, lobePruner)
+{
+    namespace sgu = MaterialXMaya::ShaderGenUtil;
+
+    auto testPath = mx::FilePath(MATERIALX_TEST_DATA);
+
+    auto library = mx::createDocument();
+    ASSERT_TRUE(library != nullptr);
+    auto searchPath = PXR_NS::HdMtlxSearchPaths();
+    mx::loadLibraries({}, searchPath, library);
+
+    auto doc = mx::createDocument();
+    doc->importLibrary(library);
+
+    auto lobePruner = sgu::LobePruner::create();
+    lobePruner->setLibrary(doc);
+
+    const auto attrVec = lobePruner->getOptimizedAttributeNames(
+        doc->getNodeDef("ND_standard_surface_surfaceshader"));
+    ASSERT_FALSE(attrVec.empty());
+    ASSERT_TRUE(std::is_sorted(attrVec.begin(), attrVec.end()));
+    ASSERT_TRUE(std::lower_bound(attrVec.begin(), attrVec.end(), "subsurface") != attrVec.end());
+
+    const auto node = doc->addNode("standard_surface", "bob", "surfaceshader");
+
+    std::string optimizedCategory;
+    ASSERT_TRUE(lobePruner->getOptimizedNodeCategory(*node, optimizedCategory));
+    // An x means can not optimize on that attribute
+    // A 0 means we optimized due to this value being zero
+    ASSERT_EQ(optimizedCategory, "standard_surface_x0000x00x000");
+
+    auto input = node->addInputFromNodeDef("subsurface");
+    input->setValueString("1.0");
+    ASSERT_TRUE(lobePruner->getOptimizedNodeCategory(*node, optimizedCategory));
+    // Now have a 1 for subsurface since we can also optimize the 1 value for mix nodes.
+    ASSERT_EQ(optimizedCategory, "standard_surface_x0000x00x010");
```

Comment:
[Minor] Do we need additional tests to cover the optimization of multiply nodes and `wight = 0` on BSDF nodes?

Reply:
Ah, perfect!